### DISCUSSION TITLE
Inextensibilty of a rod

### DISCUSSION THREAD

**Discussion by manoflogic-alt** [GazzolaLab/PyElastica#422] [Q&A]
In any code how do I ensure that the rod that I am dealing with is inextensible. For example if I apply an axial compressive force to a cantilever rod, it should twist and bend instead of reducing in length. Please help me

**Comment by armantekinalp** [comment]
Hey,

In PyElastica, it is not possible to make rods inextensible. However, you can increase the axial stiffness of the rods so that they practically behave as if they have a large resistance to extension.

To achieve this, you need to modify the `shear_matrix` of the rod. For example, to increase the axial stiffness (E * A), after creating the rod, you can scale up the shear matrix as follows:

```python
rod.shear_matrix[2,2,:] *= 10
```

This increases the rod's resistance to axial extension by 10 times.

**Reply by manoflogic-alt** [reply]
Thanks a lot
